Directory.Build.props

```diff
@@ -51,7 +51,7 @@
     <Choose>
         <When Condition="'$(NitroxLibrary)' == 'true'">
             <ItemGroup>
-                <ProjectReference Include="..\Nitrox.Analyzers\Nitrox.Analyzers.csproj" OutputItemType="Analyzer" ReferenceOutputAssembly="false" />
+                <ProjectReference Include="$(MSBuildThisFileDirectory)Nitrox.Analyzers\Nitrox.Analyzers.csproj" OutputItemType="Analyzer" ReferenceOutputAssembly="false" />
             </ItemGroup>
```

Comment:
Please check if you can use the new nuget package for our analyzers to solve the "fake" VS errors on build:
https://github.com/SubnauticaNitrox/Nitrox.Analyzers

Recommend removing this item group and changing the one below as the following:
```csproj
    <!-- Shared dependencies for all Nitrox.* projects -->
    <Choose>
        <When Condition="'$(NitroxProject)' == 'true'">
            <ItemGroup>
                <PackageReference Include="JetBrains.Annotations" Version="2022.3.1">
                    <Aliases>JB</Aliases>
                </PackageReference>
                <PackageReference Include="IsExternalInit" Version="1.0.3">
                    <PrivateAssets>all</PrivateAssets>
                    <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
                </PackageReference>
                <!-- Include our analyzer from NuGet -->
                <PackageReference Include="Nitrox.Analyzers" Version="1.0.5">
                    <PrivateAssets>all</PrivateAssets>
                    <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
                </PackageReference>
            </ItemGroup>
        </When>
    </Choose>
```

Reply:
Want me to revert my Patch method changes as well? or make a PR on the Nitrox.Analyzers Repo for the changes?

Reply:
The former, as it has less OOP complexity.

Reply:
changes made

Reply:
`override` and `TARGET_METHOD` changes are till here which aren't necessary anymore with the nuget packaged analyzers.

Reply:
Should have all other changes reverted now